dao/src/test/java/com/carhouse/dao/impl/CarSaleDaoImplTest.java

```diff
@@ -88,13 +91,24 @@ void updateCarSaleWithWrongReference() {
     @Test
     void deleteCarSale() {
         int size = carSaleDao.getCarSales().size();
-        carSaleDao.deleteCarSale(3);
+        boolean isDeleted = carSaleDao.deleteCarSale(3);
+        assertTrue(isDeleted);
         assertEquals(size - 1, carSaleDao.getCarSales().size());
         assertThrows(EmptyResultDataAccessException.class, () -> carSaleDao.getCarSale(3));
     }
 
     @Test
     void deleteCarSaleWhichHaveReferences() {
-        assertThrows(DataIntegrityViolationException.class, () -> carSaleDao.deleteCarSale(4));
+        int size = carSaleDao.getCarSales().size();
+        carSaleDao.deleteCarSale(4);
+        assertEquals(size - 1, carSaleDao.getCarSales().size());
+        EmptyResultDataAccessException thrown = assertThrows(EmptyResultDataAccessException.class,
+                () -> carSaleDao.getCarSale(4));
+        assertTrue(thrown.getMessage().contains("Incorrect result size: expected 1, actual 0"));
```

Comment:
Suppose you received this log from production environment.
Can you explain what happened?

Reply:
I don't set exception message It's spring exception
I set a clear message on the service layer
Should I remove this check from dao tests?

Reply:
Emulate this situation in app on a rest layer and add here logs. Let's try to understand will it be clear to understand what happened

Reply:
It was an incorrect method because by logick it doesn't have to throw an exception
it has to remove all references